checkov/terraform/image_referencer/base_provider.py

```diff
@@ -44,15 +46,36 @@ def extract_images_from_resources(self) -> list[Image]:
             if extract_images_func:
                 image_names.extend(extract_images_func(resource))
 
+            related_resource_id = self.get_related_resource_id(resource, file_path_to_referred_id)
             for name in image_names:
                 images.append(
                     Image(
                         file_path=resource[CustomAttributes.FILE_PATH],
                         name=name,
                         start_line=resource[START_LINE],
                         end_line=resource[END_LINE],
-                        related_resource_id=f'{removeprefix(resource.get("file_path_"), os.getenv("BC_ROOT_DIR", ""))}:{resource.get("id_")}'
+                        related_resource_id=f'{removeprefix(resource.get("file_path_"), os.getenv("BC_ROOT_DIR", ""))}:{related_resource_id}'
                     )
                 )
 
         return images
+
+    @staticmethod
+    def get_related_resource_id(resource: dict[str, Any], file_path_to_referred_id: dict[str, str]) -> str:
+        resource_id = resource.get(CustomAttributes.ID)
+        # for external modules resources the id should start with the prefix module.[module_name]
+        if resource.get(CustomAttributes.MODULE_DEPENDENCY):
+            referred_id = file_path_to_referred_id.get(f'{resource.get(CustomAttributes.FILE_PATH)}[{resource.get(CustomAttributes.MODULE_DEPENDENCY)}#{resource.get(CustomAttributes.MODULE_DEPENDENCY_NUM)}]')
+            resource_id = f'{referred_id}.{resource_id}'
+        return resource_id
+
+
+def setup_file_path_to_referred_id(graph_object: DiGraph) -> dict[str, str]:
+    file_path_to_module_id = {}
+    modules = [node for node in graph_object.nodes.values() if
+               node.get(CustomAttributes.BLOCK_TYPE) == BlockType.MODULE]
+    for modules_data in modules:
+        for module_name, module_content in modules_data.get(CustomAttributes.CONFIG, {}).items():
+            for path in module_content.get("__resolved__", []):
+                file_path_to_module_id[path] = f"module.{module_name}"
+    return file_path_to_module_id
```

Comment:
can you move this logic to `checkov/terraform/graph_builder/utils.py` in order to it in the platform instead of duplicating it?